common/djangoapps/third_party_auth/models.py

```diff
@@ -313,10 +314,20 @@ def clean(self):
         self.org_info_str = clean_json(self.org_info_str, dict)
         self.other_config_str = clean_json(self.other_config_str, dict)
 
-        self.private_key = self.private_key.replace("-----BEGIN PRIVATE KEY-----", "").strip()
-        self.private_key = self.private_key.replace("-----END PRIVATE KEY-----", "").strip()
-        self.public_key = self.public_key.replace("-----BEGIN CERTIFICATE-----", "").strip()
-        self.public_key = self.public_key.replace("-----END CERTIFICATE-----", "").strip()
+        self.private_key = (
+            self.private_key
+            .replace("-----BEGIN RSA PRIVATE KEY-----", "")
+            .replace("-----BEGIN PRIVATE KEY-----", "")
+            .replace("-----END RSA PRIVATE KEY-----", "")
+            .replace("-----END PRIVATE KEY-----", "")
```

Comment:
AFAIK, this covers the two types of output possible from the `openssl` command that's in the model's help text. For other types of key formats, they may be supported by the code that actually consumes them (python-saml). None of the code that I've written actually reads the key. This code here is just a cleanup that removes header information we don't need to store, and improves the display of the key in the admin list view.